### ISSUE TITLE
Fix Request: AvoidInfiniteRecursion false positives, call with same number of parameters

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#490]
False positive if a parameter is used in the call with calling a method on it:

```
public class Foo {
    public String translate(Request request) {
        return translate(request.getAccounts()); // good
    }
    public String translate(List<String> accounts) {
        return convert(accounts);
    }
}
```
also with an extra call in between:
```
public class Foo2 {
    public String translate(Request request) {
        return translate(Optional.ofNullable(request.getAccounts())); // good
    }
    public String translate(List<String> accounts) {
        return convert(accounts);
    }
}
```

And if any parameter is used in the call in another position:
```
public class Foo7 {
    public String translate(Request request, String str) {
        Request req;
        return translate(str, req); // good
    }
    public String translate(String str, Request request) {
        return convert(request);
    }
}
```
